File: src/hab/config/paths.py
"""Configuration file paths using platformdirs."""

from pathlib import Path

import platformdirs

APP_NAME = "home-assistant-builder"
APP_AUTHOR = "home-assistant"


def get_config_dir(custom_path: str | None = None) -> Path:
    """Get the configuration directory path.

    Args:
        custom_path: Optional custom path to use instead of default.

    Returns:
        Path to the configuration directory.
    """
    if custom_path:
        return Path(custom_path).expanduser()
    return Path(platformdirs.user_config_dir(APP_NAME, APP_AUTHOR))


def get_config_path(custom_dir: str | None = None) -> Path:
    """Get the path to config.json.

    Args:
        custom_dir: Optional custom configuration directory.

    Returns:
        Path to config.json.
    """
    return get_config_dir(custom_dir) / "config.json"


def get_credentials_path(custom_dir: str | None = None) -> Path:
    """Get the path to credentials.json.

    Args:
        custom_dir: Optional custom configuration directory.

    Returns:
        Path to credentials.json.
    """
    return get_config_dir(custom_dir) / "credentials.json"

File: config/paths.go
package config

import (
	"os"
	"path/filepath"
)

const (
	// DefaultConfigDir is the default configuration directory name
	DefaultConfigDir = "home-assistant-builder"
	// CredentialsFile is the encrypted credentials file name
	CredentialsFile = "credentials.json"
	// ConfigFile is the configuration file name
	ConfigFile = "config.json"
)

// GetConfigDir returns the configuration directory path.
// If configDir is empty, returns the default XDG config directory.
func GetConfigDir(configDir string) string {
	if configDir != "" {
		return configDir
	}

	// Try XDG_CONFIG_HOME first
	if xdgConfig := os.Getenv("XDG_CONFIG_HOME"); xdgConfig != "" {
		return filepath.Join(xdgConfig, DefaultConfigDir)
	}

	// Fall back to ~/.config
	home, err := os.UserHomeDir()
	if err != nil {
		return filepath.Join(".", DefaultConfigDir)
	}
	return filepath.Join(home, ".config", DefaultConfigDir)
}

// GetCredentialsPath returns the path to the credentials file.
func GetCredentialsPath(configDir string) string {
	return filepath.Join(GetConfigDir(configDir), CredentialsFile)
}

// GetConfigPath returns the path to the config file.
func GetConfigPath(configDir string) string {
	return filepath.Join(GetConfigDir(configDir), ConfigFile)
}

// EnsureConfigDir creates the config directory if it doesn't exist.
func EnsureConfigDir(configDir string) error {
	dir := GetConfigDir(configDir)
	return os.MkdirAll(dir, 0700)
}
